site/source/docs/porting/asyncify.rst

```diff
@@ -220,12 +220,74 @@ and want to ``await`` a dynamically retrieved ``Promise``, you can call an
 .. code-block:: cpp
 
     val my_object = /* ... */;
-    val result = my_object.call("someAsyncMethod").await();
+    val result = my_object.call<val>("someAsyncMethod").await();
 
 In this case you don't need to worry about ``ASYNCIFY_IMPORTS``, since it's an
 internal implementation detail of ``val::await`` and Emscripten takes care of it
 automatically.
 
+Note that when ASYNCIFY is used with Embind and the code is invoked from
+JavaScript then it will be implicitly treated as an async function, returning
+a ``Promise`` to the return value, as demonstrated in the following example.
+
+.. code-block:: cpp
+
+    #include <emscripten/val.h>
+    #include <emscripten/bind.h>
+    #include <stdio.h>
+
+    using namespace emscripten;
+
+    static int getAnswer(val obj) {
```

Comment:
Personally I find this example a bit hard to follow due to callbacks between JS and C++ sides.

We want to just demonstrate that async behaviour is runtime-dependant, and we can achieve that with simpler core APIs, e.g.:

```cpp
static int delayAndReturn(bool sleep, int result) {
  if (sleep) {
    emscripten_sleep(0);
  }
  return result;
}

EMSCRIPTEN_BINDINGS(example) {
  emscripten::function("delayAndReturn", &delayAndReturn);
}
```

Then, on JS side you can show what happens with just

```js
let syncResult = Module.delayAndReturn(false, 42);
console.log(syncResult); // 42

let asyncResult = Module.delayAndReturn(true, 42);
console.log(asyncResult); // Promise { <pending> }
console.log(await asyncResult); // 42
```

Reply:
Fixed.